test/e2e/upgrades/storage/volume_mode.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const devicePath = "/mnt/volume1"
+
+// VolumeModeDowngradeTest tests that a VolumeMode Block PV is not mistakenly
+// formatted and mounted like a nil/Filesystem PV after a downgrade to a version
+// where the BlockVolume feature is disabled
+type VolumeModeDowngradeTest struct {
+	pvSource *v1.PersistentVolumeSource
+	pv       *v1.PersistentVolume
+	pvc      *v1.PersistentVolumeClaim
+	pod      *v1.Pod
+}
+
+func (VolumeModeDowngradeTest) Name() string {
+	return "[sig-storage] volume-mode-downgrade"
+}
+
+func (t *VolumeModeDowngradeTest) Skip(upgCtx upgrades.UpgradeContext) bool {
+	if !framework.ProviderIs("gce", "gke") {
+		return true
+	}
+
+	// Only run when downgrading from >= 1.13 to < 1.13
+	blockVersion := version.MustParseSemantic("1.13.0")
```

Comment:
@msau42 I've written a downgrade test that tests that nothing is mapped/mounted into a pod that previously used a block volume, it doesn't seem there is a way for me to test this myself though? kubetest can downgarde from latest to e.g. v1.12.0 by doing 2 `--extract`s but I don't see a way to "--extract" my local build

Also, should I tag it Feature:ClusterDowngrade instead of Feature:VolumeModeDowngrade so that it will run with gce-master-new-downgrade-cluster?

Ty

Reply:
Yes let's set to ClusterDowngrade